scripts/rules.go

```diff
@@ -151,6 +151,29 @@ func HttpAPIErrorMessage(m dsl.Matcher) {
 		Report("Field \"Message\" should be a proper sentence with a capitalized first letter and ending in punctuation. $m")
 }
 
+// HttpAPIReturn will report a linter violation if the http function is not
+// returned after writing a response to the client.
+func HttpAPIReturn(m dsl.Matcher) {
+	m.Import("github.com/coder/coder/coderd/httpapi")
+
+	// Manually enumerate the httpapi function rather then a 'Where' condition
+	// as this is a bit more efficient.
+	m.Match(`
+	if $*_ {
+		httpapi.Write($*a)
+	}
```

Comment:
**Thought:** Does this, or should we, handle the `else` case as well? How about `switch`? Is there any way we can express that `httapi.Write` is inside a block vs function body?

Reply:
Hmm that is a good question. This is definitely the common case, but you are right about more possible situations.

I am unsure if I can match `if` OR `else` :thinking:. This does not work for example

```
$*_ {
  httpapi.Write($*a)
}
```

---

I think I will merge this in as is for now. But expanding it would be nice, I just don't think ruleguard can match different AST nodes very well. Another idea I had was trying to match the correct case like this:

```golang
m.Match(`
httpapi.Write($*a)
$r
`).Where(m["r"].Text.Matches("return"))
```

So the next line must be `return`. This is different than how I've used ruleguard in the past, but might be feasible? I was not able to get it to work with some quick tinkering. It's hard to match for the negative case (missing "return"), and when I've done similar match stuff in the past, the linter starts to become incredibly slow.

Reply:
Ah interesting, thanks for checking into it. Yup feel free to merge as-is, fixing the other cases is more of a nice-to-have anyway, def. don't want it becoming slow.